pydantic/generics.py

```diff
@@ -57,10 +61,10 @@ def __class_getitem__(cls: Type[GenericModelT], params: Union[Type[Any], Tuple[T
             ),
         )
         created_model.Config = cls.Config
-        concrete = all(not _is_typevar(v) for v in concrete_type_hints.values())
+        concrete = all(not _has_typevar(v) for v in concrete_type_hints.values())
         created_model.__concrete__ = concrete
         if not concrete:
-            parameters = tuple(v for v in concrete_type_hints.values() if _is_typevar(v))
+            parameters = tuple(v for v in concrete_type_hints.values() if _has_typevar(v))
```

Comment:
We need to extract the parameters from the concrete type hints rather than simply passing them along. Otherwise we end up with things like `Optional[T]` as a parameter.

Reply:
Nevermind. This is wrong.